### ISSUE TITLE
Tests don't pass

### ISSUE THREAD

**Issue by jtobin** [urbit/azimuth#13]
IIRC this isn't a concern (in that everything works, and we have reasons why the existing tests fail), but it's annoying.  What's up with these, and can we fix them?  Should we just delete the tests that we know fail?

**Comment by Fang-** [comment]
Wait huh, they don't? Last I checked only PlanetSale tests sporadically tripped over a tiny inaccuracy in account balance.

**Comment by jtobin** [comment]
The PlanetSale tests indeed still do that, but I also noticed some problems with NFTokenMock.  I think I used to know what the latter was related to, but don't anymore.. 😄 

Re: the PlanetSale tests, since we know about the small handful-of-wei inaccuracy and haven't judged it to be an issue, perhaps we should just relax the passing condition by comparing within +-20 wei or whatever is required (instead of a hard equality check).

**Comment by Fang-** [comment]
The NFTokenMock function should be consistent until you change the Ecliptic.sol code? iirc there was some weirdness with function overloading.

**Comment by jtobin** [comment]
Yeah I'll take a look at the NFTokenMock business.  Will reopen in the meantime.

**Comment by jtobin** [comment]
I took a look into the NFTokenMock tests and they can just fail (apparently) nondeterministically.  Sometimes they all pass, other times one or another fails due to an unhandled EVM revert.

It might be worth updating the truffle version, since there's been a new major release, and that will include a newer web3.  I'll close this one in any case.

**Comment by jtobin** [comment]
(Reopening this because the NFTokenMock test problems have annoyed me yet again, so this is clearly still an issue..)

**Comment by jtobin** [comment]
#30 may have finally fixed this.  I observed the test suite failing occasionally when I checked that branch pre-merge, but it's possible I simply forgot to `npm install` the new dependencies or something stupid like that.  I've since run the test suite a bunch of times and haven't seen any of those weird NFTokenMock issues.

I'll be bold, close this, and hope I never have to open it again.

**Comment by jtobin** [comment]
^ nm
